### PR TITLE
feat: integrate advance analytics charts

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**Description:** 
* Integrate all advance analytics plotly charts 
* Refactor to create reusable table and chart components

**JIRA:**
* https://2u-internal.atlassian.net/browse/ENT-9237
* https://2u-internal.atlassian.net/browse/ENT-9238
* https://2u-internal.atlassian.net/browse/ENT-9239

# For all changes

- [ ] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [ ] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

**Review by mahamakifdar19 — APPROVED** [review]
LGTM 👍🏻

File: src/components/AdvanceAnalyticsV2/tabs/Completions.jsx [hunk 1]
```diff
@@ -33,7 +46,26 @@ const Completions = ({
...
+            yKey: 'count',
+            colorKey: 'enrollType',
+            colorMap: chartColorMap,
+            xAxisTitle: '',
+            yAxisTitle: 'Number of Completions',
+            hovertemplate: 'Date: %{x}<br>Number of Completions: %{y}',
+          }}
+          loadingMessage={intl.formatMessage({
+            id: 'advance.analytics.completions.tab.chart.top.courses.by.completions.loading.message',
+            defaultMessage: 'Loading top courses by completions chart data',
```
Comment (jajjibhai008) [inline src/components/AdvanceAnalyticsV2/tabs/Completions.jsx:65]: Please follow the camelCase pattern to write the IDs of marked strings.

File: src/components/AdvanceAnalyticsV2/tabs/AnalyticsTable.jsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    </div>
+  );
+};
+
+AnalyticsTable.propTypes = {
+  name: PropTypes.string.isRequired,
+  tableColumns: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+  tableTitle: PropTypes.string.isRequired,
+  tableSubtitle: PropTypes.string.isRequired,
+  enableCSVDownload: PropTypes.bool.isRequired,
```
Comment (jajjibhai008) [inline src/components/AdvanceAnalyticsV2/tabs/AnalyticsTable.jsx:94]: [Suggestion] We could make it a bit more efficient by giving it a default false value. By this, we just need to pass this variable from those charts where we want to enable the download CSV button.

**Review by jajjibhai008 — APPROVED** [review]
LGTM 👏

**Pull request merged**